raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansProcessor.java

```diff
@@ -114,15 +142,20 @@ public KeyValue<TraceIdentity, StructuredTrace> transform(TraceIdentity key, Raw
     long currentTimeMs = System.currentTimeMillis();
 
     TraceState traceState = traceStateStore.get(key);
-    boolean firstEntry = (traceState == null);
-
     if (shouldDropSpan(key, traceState)) {
       return null;
     }
 
+    Event event = value.getEvent();
+    if (isMirroringSpan(event)) {
+      processMirroringSpan(key, value, traceState, currentTimeMs);
```

Comment:
we should probably call these as isPeerServiceIdenficationReuired ? or more simple enrichSpans or enrichTrace or addCustomAttributes based on use case